src/components/Table/GridTableApi.ts

```diff
@@ -175,10 +199,78 @@ export class GridTableApiImpl<R extends Kinded> implements GridTableApi<R> {
   public deleteRows(ids: string[]) {
     this.tableState.deleteRows(ids);
   }
+
+  public downloadToCsv(fileName: string): void {
+    // Create a link element, set the download attribute with the provided filename
+    const link = document.createElement("a");
+    if (link.download === undefined) throw new Error("This browser does not support the download attribute.");
+    // Create a Blob from the CSV content
+    const url = URL.createObjectURL(
+      new Blob([this.generateCsvContent().join("\n")], { type: "text/csv;charset=utf-8;" }),
+    );
+    link.setAttribute("href", url);
+    link.setAttribute("download", fileName);
+    link.style.visibility = "hidden";
+    document.body.appendChild(link);
+    link.click();
+    document.body.removeChild(link);
+  }
+
+  public copyToClipboard(): Promise<void> {
+    // Copy the CSV content to the clipboard
+    return navigator.clipboard.writeText(this.generateCsvContent().join("\n"));
```

Comment:
**nit:** This can throw an error if copying to the clipboard is not allowed. I'm not sure how this would impact the rest of the application if the error is thrown (hopefully not take it down completely), but makes me wonder if we should wrap in a try/catch and report to DataDog and maybe a `window.alert(e.message)`.

Reply:
Yeah, afaiu it would return a rejected Promise, instead of immediately throwing, so I was thinking the caller could just deal with it -- but I suppose centralized error handling would be nice, so I added a q&d window.alert for now (but still returning a rejected promise, with the idea that the caller UI could show like a red "x" instead of a green "check").

In terms of logging, I think dd should be logging all of our rejected promises, which is what this would be atm... :thinking: